### ISSUE TITLE
New-HPOVUplinkset not recognizing standard port names

### ISSUE THREAD

**Issue by sivasura** [HewlettPackard/POSH-HPOneView#186]
Please fill in as much information as possible to help resolve your issue.
When using new-HPOVuplinkSet we found that the standard format of ports given as input is not accepted.

Also tried with multiple options..
$EthernetUplinkPorts = @{
      Name = "Solution_SUS_A";
      #type = "Ethernet";
      UplinkPorts = "ENCLOSURE1:BAY1:Q1.1, ENCLOSURE1:BAY1:Q1.2, ENCLOSURE1:BAY1:Q1.3, ENCLOSURE1:BAY1:Q1.4, ENCLOSURE1:BAY1:Q2.1, ENCLOSURE1:BAY1:Q2.2, ENCLOSURE1:BAY1:Q2.3, ENCLOSURE1:BAY1:Q2.4";
      Networks = "HA_A Production_A Mgmt_A DC_Mgmt_A Migration_A";
      lacpTimer = "Short"
     }

    $Uplinkset = send-HPOVRequest -method POST -uri "/rest/uplink-sets" -body $EthernetUplinkPorts

### Expected Behavior

It should accept the standard port numbers

### Actual Behavior

PS C:\Windows\system32> New-HPOVUplinkSet -Name Solution_SUS_A -Type Ethernet -UplinkPorts "Q1.1, Q1.2, Q1.3, Q1.4, Q2.1, Q2.2, Q2.3, Q2.4"  -Networks "8G"
New-HPOVUplinkSet : Cannot validate argument on parameter 'UplinkPorts'. The 
"($_.Split(","))[0].contains(":")" validation script for the argument with value "Q1.1, Q1.2, 
Q1.3, Q1.4, Q2.1, Q2.2, Q2.3, Q2.4" did not return a result of True. Determine why the 
validation script failed, and then try the command again.
At line:1 char:68
+ ... -UplinkPorts "Q1.1, Q1.2, Q1.3, Q1.4, Q2.1, Q2.2, Q2.3, Q2.4"  -Netwo ...
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [New-HPOVUplinkSet], ParameterBindingValidationE 
   xception
    + FullyQualifiedErrorId : ParameterArgumentValidationError,New-HPOVUplinkSet

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion`):
Get-HPOVVersion
PowerShell Library Version : 3.0.1144.2035
Library Path               : C:\Program Files\WindowsPowerShell\Modules\HPOneView.300
HPE OneView Appliance Version:  

Output from `$PSVersionTable` on your Windows Host:
PS C:\Windows\system32> $PSVersionTable

Name                           Value                                                            
----                           -----                                                            
PSVersion                      5.0.10586.117                                                    
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                          
BuildVersion                   10.0.10586.117                                                   
CLRVersion                     4.0.30319.34209                                                  
WSManStackVersion              3.0                                                              
PSRemotingProtocolVersion      2.3                                                              
SerializationVersion           1.1.0.1

**Comment by ChrisLynchHPE** [comment]
Please look at the `-UplinkPorts` documented data type and format.  The value must be an `Array`, which your actual behavior is incorrect, and the second attempt is a `String` value.

It need to be

```Powershell
$EthernetUplinkPorts = @{
Name = "Solution_SUS_A";
#type = "Ethernet";
UplinkPorts = "ENCLOSURE1:BAY1:Q1.1", "ENCLOSURE1:BAY1:Q1.2", "ENCLOSURE1:BAY1:Q1.3","ENCLOSURE1:BAY1:Q1.4","ENCLOSURE1:BAY1:Q2.1","ENCLOSURE1:BAY1:Q2.2","ENCLOSURE1:BAY1:Q2.3","ENCLOSURE1:BAY1:Q2.4";
Networks = "HA_A Production_A Mgmt_A DC_Mgmt_A Migration_A";
lacpTimer = "Short"
}
```

**Comment by sivasura** [comment]
Hi Chris,
Thanks for your update. I am using the function with these options.
This works only when i have  
script line:
$task = New-HPOVUplinkSet -Inputobject $LIGobject -Name $eth -Type $nicType -UplinkPorts $nicUplinkPorts -Networks $nicName -NativeEthNetwork $NativeEth -lacpTimer $lacpTimer
value:  New-HPOVUplinkSet -Inputobject @{type=logical-interconnect-groupV300; enclosureType=C7000; snmpConfiguration=; qosConfiguration=; uplinkSets=System.Object[]; internalNetworkUris=System.Object[]; enclosureIndexes=System.Object[]; redundancyType=; interconnectMapTemplate=; stackingHealth=; telemetryConfiguration=; ethernetSettings=; interconnectBaySet=; fabricUri=/rest/fabrics/f17067d3-080d-48c5-91b5-5805cc701d5f; stackingMode=Enclosure; scopeUris=System.Object[]; description=; status=; name=Solution_LIG; state=Active; created=2016-12-23T07:44:00.670Z; modified=2016-12-23T07:44:19.132Z; eTag=28b6edf8-1571-49d9-8f26-9eef3f248610; category=logical-interconnect-groups; uri=/rest/logical-interconnect-groups/a6f1231f-bcee-4fe4-bca9-30bb57244f1d; ApplianceConnection=} -Name Solution_SUS_A -Type Ethernet -UplinkPorts se01:Bay1:Q1.1 se01:Bay1:Q2.1 -Networks Migration_A Mgmt_A HA_A DC_Mgmt_A Production_A -lacpTimer Short

This also failed with the module i used. But it was successful when i modify this line # 54624 in HPOneview.psm1.

From :  $_portRelativeValue = $_interconnecttype.portInfos | ? portName -eq $uplinkPort.Replace('.',':')
To : $_portRelativeValue = $_interconnecttype.portInfos | ? portName -eq $uplinkPort.Replace('.','.')

Please let me know if this makes sense for C7000 enclosures and if possible to do this change in the module in the next release.

Otherwise it is not working for C7000 enclosures without the enclosure hostname being part of the uplinkportid. (BAY1:Q1.1 is not accepted instead this should be Enclosure1:Q1.1) for C7000 enclosure also.


Thanks in Advance.
Regards
Siva

**Comment by ChrisLynchHPE** [comment]
No, that is not right.  The `Replace()` method is looking for the `:` character to separate out so I can figure out what Port Display Name is associated with which Interconnect Bay.  Enclosure or Frame is only required for Synergy, not C-Class.

I was able to figure out what was going on.  The problem is with how Synergy Virtual Connect Port Name changed its value from C-Class.  For instance, the following is the type of Port Name you would see in HPE Virtual Connect 20/40F8 for BladeSystem:

`Q1.1`

Now, for HPE Synergy VC 40Gb module:

`Q1:1`

The Cmdlet only supports Uplink Port Names with `Q[ParentPort].[Subport]` naming format.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 3.0.1191.2858](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v3.0.1191.2858).

Closing.

**Comment by sivasura** [comment]
Thanks Chris. This worked for my automation to create Interconnect uplinkset for C7000 Enclosures. Much appreciate your offline support as well.
Regards
Siva
